verifier-py/tests-unit/test_playbook_verifier.py

```diff
@@ -0,0 +1,261 @@
+
+    def test_true(self):
+        raw = "- bool: [true, True, TRUE]\n  string: [y, yes, Yes, YES, on, On, ON]"
+        expected = {
+            "bool": [True, True, True],
+            "string": ["y", "yes", "Yes", "YES", "on", "On", "ON"],
+        }
+
+        actual = verifier.parse_playbook(raw)
+
+        assert actual == expected
+
+
+class TestCleanPlaybook:
+    def test_ok(self):
+        raw = {
+            "name": "good playbook",
+            "hosts": "localhost",
+            "vars": {
+                "insights_signature_exclude": "/hosts,/vars/insights_signature/",
+                "insights_signature": b"data",
+            },
+            "tasks": [],
+        }
+        expected = {
+            "name": "good playbook",
+            "vars": {"insights_signature_exclude": "/hosts,/vars/insights_signature/"},
+            "tasks": [],
+        }
+
+        actual: dict = verifier.clean_playbook(raw)
+        assert actual == expected
+
+    def test_requires_vars(self):
+        raw = {"name": "bad playbook", "tasks": [{"name": "a task"}]}
+
+        with pytest.raises(Exception, match="does not have the key 'vars"):
+            verifier.clean_playbook(raw)
+
+    def test_requires_signature_exclude(self):
+        raw = {"name": "bad playbook", "vars": {}, "tasks": [{"name": "a task"}]}
+
+        with pytest.raises(
+            Exception, match="does not have the key 'vars/insights_signature_exclude'"
+        ):
+            verifier.clean_playbook(raw)
+
+    def test_too_shallow_exclude(self):
+        raw = {"vars": {"insights_signature_exclude": "/"}}
+
+        with pytest.raises(Exception, match="too deep or shallow"):
+            verifier.clean_playbook(raw)
+
+    def test_too_deep_exclude(self):
+        raw = {"vars": {"insights_signature_exclude": "/vars/nested/key"}}
+
+        with pytest.raises(Exception, match="too deep or shallow"):
+            verifier.clean_playbook(raw)
+
+    def test_forbidden_exclude(self):
+        raw = {"vars": {"insights_signature_exclude": "/name"}}
+
+        with pytest.raises(Exception, match="cannot be excluded"):
+            verifier.clean_playbook(raw)
+
+    def test_missing_simple(self):
+        raw = {"vars": {"insights_signature_exclude": "/hosts"}}
+
+        with pytest.raises(Exception, match="field '/hosts' is not present in the playbook"):
+            verifier.clean_playbook(raw)
+
+    def test_missing_nested(self):
+        raw = {"vars": {"insights_signature_exclude": "/vars/insights_signature"}}
+
+        with pytest.raises(
+            Exception, match="field '/vars/insights_signature' is not present in the playbook"
+        ):
+            verifier.clean_playbook(raw)
+
+
+class TestPlaybookDigest:
+    def test_ok(self):
+        playbooks_dir = pathlib.Path(__file__).parents[2] / "data" / "playbooks"
+
+        with (playbooks_dir / "insights_remove.serialized.bin").open("rb") as f:
+            raw: bytes = f.read()
```

Comment:
this and the other occurrences in this file: maybe use the `read_bytes()` method of a `Path`? eg
```python
        raw = (playbooks_dir / "insights_remove.serialized.bin").read_bytes()
```